docs/tutorial/quickstart.md

```diff
@@ -0,0 +1,355 @@
+
+```
+package main
+
+import (
+	"fmt"
+
+	"oras.land/oras-go/v2/registry/remote"
+	"oras.land/oras-go/v2/registry/remote/auth"
+	"oras.land/oras-go/v2/registry/remote/retry"
+)
+
+func main() {
+	// 1. Connect to a remote repository with basic authentication
+	registry := "example.registry.com"
+	repository := "myrepo"
+	repo, err := remote.NewRepository(fmt.Sprintf("%s/%s", registry, repository))
+	if err != nil {
+		panic(err)
+	}
+	// Note: The below code can be omitted if authentication is not required.
+	repo.Client = &auth.Client{
+		Client: retry.DefaultClient,
+		Cache:  auth.NewCache(),
+		Credential: auth.StaticCredential(registry, auth.Credential{
+			Username: "username",
+			Password: "password",
+		}),
+	}
+}
+```
+
+## Show tags in the repository
+The following code snippet uses the [(*Repository) Tags](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Tags) method from the [remote](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote) package to list the tags in the repository. Paste the code into `main.go` after the last section.
+
+```
+// 2. Show the tags in the repository
+ctx := context.Background()
+err = repo.Tags(ctx, "", func(tags []string) error {
+	for _, tag := range tags {
+		fmt.Println(tag)
+	}
+	return nil
+})
+if err != nil {
+	panic(err)
+}
+```
+
+### Run the code
+
+At the command line, use [`go get`](https://golang.org/cmd/go/#hdr-Add_dependencies_to_current_module_and_install_them) to add the `oras.land/oras-go/v2` module as a dependency for your module. Use a dot argument to mean "get dependencies for code in the current directory."
+
+```
+go get .
+go: added github.com/opencontainers/go-digest v1.0.0
+go: added github.com/opencontainers/image-spec v1.1.0
+go: added golang.org/x/sync v0.6.0
+go: added oras.land/oras-go/v2 v2.5.0
+```
+
+Run the code.
+```
+go run .
+```
+
+You should see the tags in the repository.
+
+## Push a layer to the repository
+
+All referenced layers must exist in the repository before a manifest can be pushed. The following code snippet demonstrates how to push a manifest layer using the [(*Repository) Push](https://pkg.go.dev/oras.land/oras-go/v2@v2.5.0/registry/remote#Repository.Push) method. Paste the code into `main.go` after the last section.
+
+```
+// 3. push a layer to the repository
+layer := []byte("example manifest layer")
+layerDescriptor := content.NewDescriptorFromBytes(v1.MediaTypeImageLayer, layer)
+err = repo.Push(ctx, layerDescriptor, bytes.NewReader(layer))
+if err != nil {
+	panic(err)
+}
```

Comment:
It's too complex. Consider [PushBytes](https://pkg.go.dev/oras.land/oras-go/v2#PushBytes).